### PR TITLE
Feature#51/mailer config

### PR DESCRIPTION
**Pull request by juperameza**
# Description
 - What?: I've added the option for the user to reset their password using devise and mailtrap.
 - Why?: To implement the first configuration .
 See more information at Issue #51 
 # Testing
You can see the preview at http://localhost:3000/rails/mailers/user_mailer
And can also test it running rails devise:send_reset_password_instructions_to_all_users it should print the id of the users

### CODE REVIEW

**Review by AlejandroCuriel — DISMISSED** [review]
LGTM MACHIN :full_moon:

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -24,10 +24,10 @@
   # Configure the e-mail address which will be shown in Devise::Mailer,
   # note that it will be overwritten if you use your own mailer class
   # with default "from" parameter.
-  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
+   config.mailer_sender = 'donotreply@example.com'
 
   # Configure the class responsible to send e-mails.
-  # config.mailer = 'Devise::Mailer'
+   config.mailer = 'UserMailer'
```
Comment (dannharo) [inline config/initializers/devise.rb:31]: the indentation was the correct before the change

File: lib/tasks/devise.rake [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    desc "Send welcome reset password instructions to all users.
+    This will lockout all users until they reset their password."
+    task send_reset_password_instructions_to_all_users: :environment do
+        User.where.not(email: nil).in_batches(of: 10).each_record do |user|
+          # Send instructions so user can enter a new password:
+          UserMailer.reset_password_instructions(user).deliver
+          p user.id
+        end
+     end
+  end
```
Comment (dannharo) [inline lib/tasks/devise.rake:11]: check your indentation here also the next file `devise.rb`

File: test/mailers/previews/user_mailer_preview.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+
+# Preview all emails at http://localhost:3000/rails/mailers/user_mailer
+class UserMailerPreview < ActionMailer::Preview
+    def reset_password_instructions
+        UserMailer.reset_password_instructions(User.first)
```
Comment (dannharo) [inline test/mailers/previews/user_mailer_preview.rb:5]: also indentation here

**Review by dannharo — COMMENTED** [review]
Looks good
as recommendation you could use [editorconfig](https://editorconfig.org/) to help you with some configurations including the indentation

File: app/views/users/mailer/reset_password_instructions.html.erb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    <title>Reestablecer contraseña para local-ecommerce</title>
+  </head>
+  <body>
+    <div>
+      <p>Hola,</p>
+      <p>Recibimos tu solicitud para reestablecer tu contraseña</p>
+      <p> <%=link_to "Presiona este enlace para reestablecer tu contraseña", edit_user_password_url(reset_password_token: @token), html_options = { style: "font-weight: bold" } %>.</p>
+      <p>Tienes 6 horas para usar este enlace, si no solicitaste un cambio de contraseña ignora este correo</p>
+      <p> Si tienes algun problema contacta a  <a href="mailto:alex_curielg@hotmail.com">support@example.com</a>. Y nuestro equipo te ayudara.</p>
+      <p>TNo compartas este link con nadie!</p>
```
[the remark this answers is not in the document]
Comment (juperameza) [reply app/views/users/mailer/reset_password_instructions.html.erb]: Done

File: app/mailers/user_mailer.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+    private
+    def create_reset_password_token(user)
+        raw, hashed = Devise.token_generator.generate(User, :reset_password_token)
+        @token = raw
+        user.reset_password_token = hashed
+        user.reset_password_sent_at = Time.now.utc
+        user.save
+    end
+end
```
[the remark this answers is not in the document]
Comment (juperameza) [reply app/mailers/user_mailer.rb:20]: Done

**Review by germanruzca — APPROVED** [review]
LGTM

**Review by germanruzca — APPROVED** [review]
LGTM

**Review by EliasUR — APPROVED** [review]
LGTM

**Review by germanruzca — APPROVED** [review]

**Pull request merged**